tests/test_authz_authorizer.py

```diff
@@ -38,37 +32,161 @@ def test__init__(self):
         assert self.authz.resource == "test_resource"
         assert self.authz.permission == "permission_name"
 
-    def test__repr__(self):
+    def test__repr__(self) -> None:
         assert repr(self.authz) == (
             "Authorizer(auth_jwt=<jwt>, "
             "resource_name='test_resource', "
             "permission_name='permission_name')"
         )
 
-    def test__set_policy_w_scope(self):
+    def test_validate_one_with_expired(self) -> None:
+        expiration_negative = int((datetime.utcnow() - timedelta(seconds=1)).timestamp())
+        with pytest.raises(Unauthorized):
+            self._make_authorizer(
+                {
+                    **self.token_payload,
+                    "allow": {"test_resource": {"permission_name": {"allow": ALL}}},
+                    "exp": expiration_negative,
+                }
+            )
+
+
+# pylint: disable=too-many-public-methods
+class TestAuthorizerSetupMethod:
+    @staticmethod
+    def _mocked_make_authorizer(token_payload: dict) -> Authorizer:
+        with patch("lbz.authz.authorizer.decode_jwt", lambda _: token_payload):
+            return Authorizer("xx", "test_resource", "permission_name")
+
+    @pytest.fixture(autouse=True)
+    def setup_method(self, base_auth_payload) -> None:
+        # pylint: disable=attribute-defined-outside-init
+        self.token_payload = base_auth_payload
+        self.authz = self._mocked_make_authorizer(self.token_payload)
+
+    def test_wrong_jwt_authz_payload_raises_permission_denied(self) -> None:
+        with pytest.raises(PermissionDenied):
+            self._mocked_make_authorizer({})
+
+    def test_check_deny_res(self) -> None:
```

Comment:
_FYI:_ Changing the order of the tests when you are making multiple "fixes" in-between makes review much much harder.